### ISSUE TITLE
Is it possible to cancel upgrade/unit ?

### ISSUE THREAD

**Issue by DevJhns** [Dentosal/python-sc2#59]
So I'm interested if it possible to cancel, for example stalker in gateway, 
and does build_progress can work with upgrade/unit ?

**Comment by mjschuetze102** [comment]
Yes, it is possible. I have done it with extractors to get the drone back. You will need to call the CANCEL Ability on the gateway that is creating the stalker. It should look similar to something like this:

```
extractor = self.units(UnitTypeId.EXTRACTOR).not_ready.first
await self.do(extractor(AbilityId.CANCEL))
```

**Comment by DevJhns** [comment]
@mjschuetze102  wow, didn't noticed this in docs, thx a lot! 
and what about ability\upgrade  research progress isit possible?

**Comment by mjschuetze102** [comment]
I haven't looked into this. I would assume something like self.already_pending(research) would work. Again, I wouldn't know though. You can see if it has been completely researched by selecting a unit that would have the upgrade and seeing if the abilities on it have changed. Or by selecting the building doing the research and seeing if it is still available or not. But neither of those would give you how far along the research is. #46 may be what you're looking for.
